### PR TITLE
Add OPML file parsing

### PR DESCRIPTION
**Pull request by AngeloStavrow**
This PR adds an `/opml` endpoint in **server.js** which uses **opmlparser.js** to:

1. parse the OPML file at `public/subscriptions.opml`;
2. send each feed URL to the `/api/parse/:feed` endpoints; and
3. return the JSON all feeds to the client.

You can test this by going to https://rss-firehose-parseopml.glitch.me/opml and checking out the returned JSON for the Glitch feed from Dev.to and my own feed on Dev.to — it's helpful to have a JSON viewer extension installed in your browser (Firefox Developer Edition handles JSON nicely by default).

### CODE REVIEW

File: opmlparser.js [hunk 1]
```diff
@@ -0,0 +1,46 @@
+// Adapted from https://github.com/danmactough/node-opmlparser/blob/master/examples/simple.js
+
+const OpmlParser = require("opmlparser");
+const request = require("request");
+
+exports.getFeedsFromOPML = () =>
+  new Promise((resolve, reject) => {
+    const opmlparser = new OpmlParser()
+    let counter = 0;
```
Comment (AngeloStavrow) [inline opmlparser.js:9]: I don't think this needs to be here.

File: server.js [hunk 2]
```diff
@@ -51,6 +52,22 @@ app.get("/", (request, response) => {
   response.sendFile(`${__dirname}/views/index.html`);
 });
 
+app.get("/opml", async function(request, response) {
```
Comment (melissamcewen) [inline server.js:56]: Add comment

File: opmlparser.js [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+      this.pipe(opmlparser);
+    })
+
+    opmlparser.on("error", function(error) {
+      // Reject the Promise by returning an error, with the origin
+      // of the error set to the request.
+      reject({ error: error });
+    });
+  
+    opmlparser.on("readable", function () {
```
Comment (melissamcewen) [inline opmlparser.js:33]: Add comment

**Review by melissamcewen — CHANGES_REQUESTED** [review]
Looks great. There were a few areas I noticed no comment that I noted here, they just stood out because most other code was commented.

**Pull request merged**